src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -625,12 +622,30 @@ static byte[] validateAndParseBinary(
       output = new byte[originalInputArray.length];
       System.arraycopy(originalInputArray, 0, output, 0, originalInputArray.length);
     } else if (input instanceof String) {
-      try {
-        String stringInput = ((String) input).trim();
-        output = Hex.decodeHex(stringInput);
-      } catch (DecoderException e) {
+      if(binaryStringEncoding == Constants.BinaryStringEncoding.BASE64) {
+        try {
+          String stringInput = ((String) input).trim();
+          // Remove double quotes if present
```

Comment:
We should not strip surrounding quotes. We don't do it for hex and neither does it work in Snowflake Worksheets.